tests/test_data.py

```diff
@@ -454,6 +454,33 @@ def test_get_data():
     assert type(data) == io.BytesIO
 
 
+def test_masked_data_mutable():
+    with pm.Model():
+        data = np.ma.MaskedArray([1.0, 2.0, 3], [0, 0, 1])
+        expected = np.array([1, 2, np.nan])
+        with pytest.warns(UserWarning, match="masked arrays"):
+            result = pm.MutableData("test", data).get_value()
+        np.testing.assert_array_equal(result, expected)
+
+
+def test_masked_data_constant():
+    with pm.Model():
+        data = np.ma.MaskedArray([1.0, 2.0, 3], [0, 0, 1])
+        expected = np.array([1, 2, np.nan])
+        with pytest.warns(UserWarning, match="masked arrays"):
+            result = pm.ConstantData("test", data).data
+        np.testing.assert_array_equal(result, expected)
+
+
+def test_masked_integer_data():
+    with pm.Model():
+        data = np.ma.MaskedArray([1, 2, 3], [0, 0, 1])
```

Comment:
Integers should be fine, otherwise we can't input discrete variables?

Reply:
Unfortunately, we can't - you cannot have an integer NumPy array with `nan` values, ie. this throws an error: `np.array([1,2,3, np.nan[. dtype=int)`. That's because `nan` is strictly a float concept. So yes, we would not be able to allow users pass an integer masked array into `pm.Data()`. If they want to benefit from automatic imputation, they can today (and will be after this PR) pass a masked integer array directly into `observed` parameter of an RV. I have an error message that explains the options in the code: 

https://github.com/pymc-devs/pymc/pull/6645/commits/28d15f809ddfa8d5d490e30107194d2d1b7d7228#diff-823b37f218229d363550b4cc387cfffa180c5c6e0e5ad0e174f2f0be7aa4692aR102

```
if isinstance(data, np.ma.MaskedArray):
        if "int" in str(data.dtype):
            raise TypeError(
                "Masked integer arrays (integer type datasets with missing values) are not supported by pm.Data() / pm.Model.set_data() at this time. \n"
                "Consider if using a float type fits your use case. \n"
                "Alternatively, if you want to benefit from automatic imputation in pyMC, pass a masked array directly to `observed=` parameter when defining a distribution."
            )
        else:
            ret = data.filled(fill_value=np.nan)
```

Reply:
I wasn't clear. Is any error raised if user pass floats observed to discrete variables? I think it works just fine.